HW05.Rmd

```diff
@@ -0,0 +1,159 @@
+---
+title: "HW05"
+author: "Xing Yanfeng"
+date: "2020/8/2"
+output: github_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+## Assignment 1
+
+### Load libraries and read in data.
+
+```{r libraries data}
+# load libraries
+library(tidyr)
+library(dplyr)
+library(readr)
+library(ggplot2)
+# read in data
+dad_mom <- read_delim("dad_mom.txt", delim = "\t")
+```
+
+### Tidying data
+
+```{r tidying}
+tidy_dad_mom <- dad_mom %>% 
+  gather(key = "key", value = "value", 2:5) %>% 
+  separate(col = key, into = c("key", "relationship")) %>% 
+  spread(key = key, value = value, convert = T) %>% 
+# rearrange columns to make it more comfortable
+  select(fam_id, relationship, name, income)
+print(tidy_dad_mom)
```

Comment:
Well executed data clean-up. I don't have much more to add since I've done it very similarly. I like the use of select() to reorder columns.